### PR TITLE
Feat(eos_cli_config_gen): Add advertise_map and supress_map keys to BGP template

### LINKED ISSUE #3316: Add support for route aggregates and as-path ACLs
### Enhancement summary

Add support for the following CLI:

```
ip as-path access-list AS-PATH-ACL-NAME seq 5 permit "^1234_[1-9]+"
!
route-map MY-RM permit 10
  match as-path AS-PATH-ACL-NAME
!
router bgp 100
  vrf VRF-NAME
  aggregate-address 1.0.0.0/8 as-set advertise-map MY-RM suppress-map MY-RM
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Need to configure route aggregates and AS-PATH ACLs to be mapped to route-map entries as part of EOS deployments with AVD

### Describe the solution you would like

Add J2 template for as-path ACLS, and extend router_bgp template and route-map template if required to add support for mapping as-path ACLs

### Describe alternatives you have considered

eos_cli or custom template

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gusmb**
## Change Summary

Add the `advertise_map` and `supress_map` parameters to the BGP templates to cover the CLI specified on related issue

## Related Issue(s)

Fixes #3316 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Add CLI parameters to templates to support the following CLI:

```
router bgp 100
  vrf VRF-NAME
  aggregate-address 1.0.0.0/8 as-set advertise-map MY-RM suppress-map MY-RM
```

## How to test

See molecule updates

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by carlbuchmann** [comment]
Moving to draft since PR is failing pre-commit and docs need to be generated.

**Review by ClausHolbechArista — APPROVED** [review]

File: python-avd/pyavd/schema/utils.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
```
Comment (Shivani-gslab) [inline python-avd/pyavd/schema/utils.py]: Is this file required?

Comment (ClausHolbechArista) [reply python-avd/pyavd/schema/utils.py]: Good point. Yes it was a mistake on my end.

**Review by Shivani-gslab — APPROVED** [review]
LGTM!
